switch_mod/solve.py

```diff
@@ -1,62 +1,398 @@
-# Copyright 2015 The Switch Authors. All rights reserved.
-# Licensed under the Apache License, Version 2, which is in the LICENSE file.
+#!/usr/bin/env python
 
-"""
-Command line front-end for running the Switch model solver.
+import sys, os, time, traceback, shlex, re
 
-Usage:  python -m switch_mod.solve [ARGS]
-"""
+from pyomo.environ import *
+from pyomo.opt import SolverFactory, SolverStatus, TerminationCondition
 
-import argparse
-import os
-import sys
+def add_relative_path(*parts):
+    """ Adds a new path to sys.path.
+    The path should be specified relative to the current module,
+    and specified as a list of directories to traverse to reach the
+    final destination."""
+    sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), *parts)))
 
-import pyomo.opt
+add_relative_path('switch')   # standard switch model
+from switch_mod.utilities import create_model, _ArgumentParser
 
-import switch_mod.utilities
+add_relative_path('.') # components for this particular study
 
+def main(args=sys.argv[1:]):
+        
+    # build a module list based on configuration options passed in,
+    # and add the current module (to register define_arguments callback)
+    modules = get_module_list(args)
+    
+    # NOTE: we have (at least) two options for parsing model configuration arguments 
```

Comment:
You can ignore these notes (we should delete most of them soon). The approach I ended up adopting was to put  minimal support for arguments in switch_mod.utilities (create an argument parser, pass it to the modules, parse the arguments and put the results in model.options), and then have each module define and consume the arguments it needs. So switch_mod.utilities doesn't actually define or operate on any arguments, but it provides a skeleton framework which makes it easy to use them everywhere.